hadoop-etl-common/src/test/java/com/exasol/hadoop/HdfsSerDeImportServiceDateTimeTest.java

```diff
@@ -0,0 +1,126 @@
+        List<HCatSerDeParameter> serDeParameters = new ArrayList<>(
+                Arrays.asList(new HCatSerDeParameter("serialization.format", "1")));
+
+        String inputFormatClassName = "org.apache.hadoop.hive.ql.io.RCFileInputFormat";
+        String serDeClassName = "org.apache.hadoop.hive.serde2.columnar.ColumnarSerDe";
+
+        List<String> hdfsServers = new ArrayList<>(Arrays.asList("file:///"));
+
+        InputFormat<?, ?> inputFormat =
+            (InputFormat<?, ?>) UdfUtils.getInstanceByName(inputFormatClassName);
+        AbstractSerDe serDe = (AbstractSerDe) UdfUtils.getInstanceByName(serDeClassName);
+        HdfsSerDeImportService.importFile(fileSystem, file, partitionColumns, inputFormat, serDe,
+                serDeParameters, hdfsServers, "hdfs", columns, outputColumns, false, false, ctx);
+    }
+
+    @Test
+    public void testImportDateTypeAsSQLDate() throws Exception {
+        final RCFileWriter.Type type = RCFileWriter.Type.DATE;
+        final RCFileWriter rcWriter = new RCFileWriter(conf, fileSystem, type);
+
+        final List<Date> values = new ArrayList<>();
+        int count = 0;
+        for (int i = 31230; i <= 31234; i++) {
+            count++;
+            values.add(Date.ofEpochDay(i));
+        }
+        rcWriter.writeValues(temporaryFile, values);
+
+        ExaIterator ctx = mock(ExaIterator.class);
+        final List<HCatTableColumn> columns = Arrays.asList(
+                new HCatTableColumn(type.getColumnName(), type.getColumnType()));
+        final List<OutputColumnSpec> outputColumns = OutputColumnSpecUtil
+            .generateDefaultOutputSpecification(columns, Collections.emptyList());
+
+        runImportRCFile(ctx, columns, Collections.emptyList(), outputColumns,
+                temporaryFile.toURI().toString());
+
+        verify(ctx, times(count)).emit(anyVararg());
+        for (final Date date : values) {
+            verify(ctx, times(1)).emit(eq(new java.sql.Date(date.toEpochMilli())));
+        }
+    }
+
+    @Test
+    public void testImportTimestampTypeAsSQLTimestamp() throws Exception {
+        final RCFileWriter.Type type = RCFileWriter.Type.TIMESTAMP;
+        final RCFileWriter rcWriter = new RCFileWriter(conf, fileSystem, type);
+
+        final List<Timestamp> values = new ArrayList<>();
+        int count = 0;
+        for (int i = 31230; i <= 31234; i++) {
+            count++;
+            values.add(Timestamp.ofEpochMilli(i));
+        }
+        rcWriter.writeValues(temporaryFile, values);
+
+        ExaIterator ctx = mock(ExaIterator.class);
+        final List<HCatTableColumn> columns = Arrays.asList(
+                new HCatTableColumn(type.getColumnName(), type.getColumnType()));
+        final List<OutputColumnSpec> outputColumns = OutputColumnSpecUtil
+            .generateDefaultOutputSpecification(columns, Collections.emptyList());
+
+        runImportRCFile(ctx, columns, Collections.emptyList(), outputColumns,
```

Comment:
These two methods are very similar. Any chances to extract a common part to a method?

Reply:
Extracted common parts.